src/main/scala/edu/berkeley/ce/rockslicing/JointGenerator.scala

```
@@ -0,0 +1,221 @@
+    *                 first called.
+    * @return Seq of joints representing the joint set
+    */
+  @tailrec
+  private def makePersistentJointSet(joint: Joint, jointData: Array[Double], diagonalVector: DenseVector[Double],
+                                     center: Array[Double], terminationPoint: Array[Double],
+                                     jointSet: Seq[Joint]): Seq[Joint] = {
+    val terminationVector = DenseVector[Double](terminationPoint(0) - center(0),
+      terminationPoint(1) - center(1),
+      terminationPoint(2) - center(2)
+    )
+    // Check if center is still located within bounding box
+    if (terminationVector.dot(diagonalVector) >= -NumericUtils.EPSILON) {
+      // Create new joint with updated center
+      val newJoint = Joint(Array(joint.a, joint.b, joint.c), origin, center, joint.phi, joint.cohesion,
+        joint.shape, joint.dipAngleParam, joint.dipDirectionParam)
+      // Update center based on joint spacing and diagonal vector
+      val centerIncrement = jointData(2) / math.abs(diagonalVector dot DenseVector[Double](joint.a, joint.b, joint.c))
+      val newCenter = Array(
+        diagonalVector(0)*centerIncrement + center(0),
+        diagonalVector(1)*centerIncrement + center(1),
+        diagonalVector(2)*centerIncrement + center(2)
+      )
+      makePersistentJointSet(joint, jointData, diagonalVector, newCenter, terminationPoint, newJoint +: jointSet)
+    } else {
+      jointSet
+    }
+  }
+
+  /**
+    * This function is not yet implemented. Placeholder until verifying current load balance logic works.
+    *
+    * @param joint
+    * @param jointData
+    * @return
+    */
+  def makeNonPersistentJointSet(joint: Joint, jointData: Array[Double]): Seq[Joint] = {
+    // Still needs to be implemented, but will require stochastic generator
+    println("ERROR: Non-persistent joint generator not yet implemented")
```

Let's throw an `UnsupportedOperationException` here.